configs/ds_pretrain_gpt2.yml

```diff
@@ -1,42 +1,42 @@
 {
-   "num_nodes":"1",
-   "num_gpus":"8",
-   "model-parallel-size":"4",
-   "num-layers":"24",
-   "hidden-size":"1024",
-   "num-attention-heads":"16",
-   "seq-length":"1024",
-   "max-position-embeddings":"1024",
-   "batch-size":"9",
-   "train-iters":"320000",
-   "lr-decay-iters":"320000",
-   "save":"checkpoints/gpt2_345m_ds",
-   "load":"checkpoints/gpt2_345m_ds",
-   "data-impl":"mmap",
-   "split":"949,50,1",
-   "distributed-backend":"nccl",
-   "lr":"1.5e-4",
-   "lr-decay-style":"cosine",
-   "min-lr":"1.0e-5",
-   "weight-decay":"1e-2",
-   "clip-grad":"1.0",
-   "warmup":"0.01",
-   "checkpoint-activations":true,
-   "log-interval":"100",
-   "save-interval":"10000",
-   "eval-interval":"1000",
-   "eval-iters":"10",
-   "tensorboard-dir":"tensorboard_data/24l_1024h_1n_8g_4mp_9b_ds4",
-   "deepspeed":true,
-   #"deepspeed_config":"configs/deepspeed_configs/ds_config.json",
-   "zero-stage":"0",
-   "zero-reduce-bucket-size":"50000000",
-   "zero-allgather-bucket-size":"5000000000",
-   "zero-contigious-gradients":true,
-   "zero-reduce-scatter":true,
-   "checkpoint-num-layers":"1",
-   "partition-activations":true,
-   "synchronize-each-layer":true,
+   "num_nodes": 1,
+   "num_gpus": 8,
+   "model-parallel-size": 4,
+   "num-layers": 24,
+   "hidden-size": 1024,
+   "num-attention-heads": 15,
+   "seq-length": 1024,
+   "max-position-embeddings": 1024,
+   "batch-size": 9,
+   "train-iters": 320000,
+   "lr-decay-iters": 320000,
+   "save": "checkpoints/gpt2_345m_ds",
+   "load": "checkpoints/gpt2_345m_ds",
+   "data-impl": "mmap",
+   "split": "949,50,1",
+   "distributed-backend": "nccl",
+   "lr": 1.5e-4,
+   "lr-decay-style": "cosine",
+   "min-lr": 1.0e-5,
+   "weight-decay": 1e-2,
+   "clip-grad": 1.0,
+   "warmup": 0.01,
+   "checkpoint-activations": True,
```

Comment:
@StellaAthena `true` is how true is represented in JSON. YAML allows for either `True` or `true` but lets just use `true/false` for now

Reply:
Hmm interesting. I had assumed consistency with Python was preceded, but if JSONs use a different standard I’m happy to use what’s consistent with that.